### PR TITLE
Standardize integer type checks

### PR DESCRIPTION
**Pull request by achabotl**
#### What does this implement/fix? Explain your changes.

Adds an `is_positive_int` function to `util` and uses it wherever I could spot `isinstance` and `issubdtype` checks for positive integer input argument checks.

Fixes: #1519.

### CODE REVIEW

**Review by bmcfee — APPROVED** [review]
All LGTM, thanks!  Only thing that's missing is adding `is_positive_int` to `util.__init__`'s autodoc index.

File: librosa/core/audio.py [hunk 1]
```diff
@@ -415,11 +415,11 @@ def stream(
 
     """
 
-    if not (np.issubdtype(type(block_length), np.integer) and block_length > 0):
+    if not util.is_positive_int(block_length):
         raise ParameterError("block_length={} must be a positive integer")
```
Comment (achabotl) [inline librosa/core/audio.py:419]:
```suggestion
        raise ParameterError(f"block_length={} must be a positive integer")
```

File: librosa/core/audio.py [hunk 2]
```diff
@@ -415,11 +415,11 @@ def stream(
...
-    if not (np.issubdtype(type(frame_length), np.integer) and frame_length > 0):
+    if not util.is_positive_int(frame_length):
         raise ParameterError("frame_length={} must be a positive integer")
```
Comment (achabotl) [inline librosa/core/audio.py:421]:
```suggestion
        raise ParameterError(f"frame_length={} must be a positive integer")
```

File: librosa/core/audio.py [hunk 3]
```diff
@@ -415,11 +415,11 @@ def stream(
...
-    if not (np.issubdtype(type(hop_length), np.integer) and hop_length > 0):
+    if not util.is_positive_int(hop_length):
         raise ParameterError("hop_length={} must be a positive integer")
```
Comment (achabotl) [inline librosa/core/audio.py:423]:
```suggestion
        raise ParameterError(f"hop_length={} must be a positive integer")
```

File: librosa/sequence.py [hunk 4]
```diff
@@ -1467,7 +1467,7 @@ def transition_uniform(n_states):
            [0.333, 0.333, 0.333]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 0:
+    if not is_positive_int(n_states):
         raise ParameterError("n_states={} must be a positive integer")
```
Comment (achabotl) [inline librosa/sequence.py:1471]:
```suggestion
        raise ParameterError(f"n_states={} must be a positive integer")
```

File: librosa/sequence.py [hunk 5]
```diff
@@ -1515,7 +1515,7 @@ def transition_loop(n_states, prob):
            [0.375, 0.375, 0.25 ]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 1:
+    if not (is_positive_int(n_states) and (n_states > 1)):
         raise ParameterError("n_states={} must be a positive integer > 1")
```
Comment (achabotl) [inline librosa/sequence.py:1519]:
```suggestion
        raise ParameterError(f"n_states={} must be a positive integer > 1")
```

File: librosa/sequence.py [hunk 6]
```diff
@@ -1582,7 +1582,7 @@ def transition_cycle(n_states, prob):
            [0.1, 0. , 0. , 0.9]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 1:
+    if not (is_positive_int(n_states) and n_states > 1):
         raise ParameterError("n_states={} must be a positive integer > 1")
```
Comment (achabotl) [inline librosa/sequence.py:1586]:
```suggestion
        raise ParameterError(f"n_states={} must be a positive integer > 1")
```

File: librosa/sequence.py [hunk 7]
```diff
@@ -1683,7 +1683,7 @@ def transition_local(n_states, width, *, window="triangle", wrap=False):
            [0.   , 0.   , 0.   , 0.   , 1.   ]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 1:
+    if not (is_positive_int(n_states) and n_states > 1):
         raise ParameterError("n_states={} must be a positive integer > 1")
```
Comment (achabotl) [inline librosa/sequence.py:1687]:
```suggestion
        raise ParameterError(f"n_states={} must be a positive integer > 1")
```

**Comment by achabotl** [comment]
Ready for re-review and merge. Executable bit removed, added to auto doc, added f-strings.

File: librosa/core/audio.py [hunk 8]
```diff
@@ -415,12 +415,12 @@ def stream(
...
     """
 
-    if not (np.issubdtype(type(block_length), np.integer) and block_length > 0):
-        raise ParameterError("block_length={} must be a positive integer")
-    if not (np.issubdtype(type(frame_length), np.integer) and frame_length > 0):
-        raise ParameterError("frame_length={} must be a positive integer")
-    if not (np.issubdtype(type(hop_length), np.integer) and hop_length > 0):
-        raise ParameterError("hop_length={} must be a positive integer")
+    if not util.is_positive_int(block_length):
+        raise ParameterError(f"block_length={} must be a positive integer")
```
Comment (bmcfee) [inline librosa/core/audio.py:419]:
```suggestion
        raise ParameterError(f"block_length={block_length} must be a positive integer")
```

File: librosa/core/audio.py [hunk 9]
```diff
@@ -415,12 +415,12 @@ def stream(
...
+    if not util.is_positive_int(frame_length):
+        raise ParameterError(f"frame_length={} must be a positive integer")
```
Comment (bmcfee) [inline librosa/core/audio.py:421]:
```suggestion
        raise ParameterError(f"frame_length={frame_length} must be a positive integer")
```

File: librosa/core/audio.py [hunk 10]
```diff
@@ -415,12 +415,12 @@ def stream(
...
+    if not util.is_positive_int(hop_length):
+        raise ParameterError(f"hop_length={} must be a positive integer")
```
Comment (bmcfee) [inline librosa/core/audio.py:423]:
```suggestion
        raise ParameterError(f"hop_length={hop_length} must be a positive integer")
```

File: librosa/sequence.py [hunk 11]
```diff
@@ -1467,8 +1467,8 @@ def transition_uniform(n_states):
            [0.333, 0.333, 0.333]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 0:
-        raise ParameterError("n_states={} must be a positive integer")
+    if not is_positive_int(n_states):
+        raise ParameterError(f"n_states={} must be a positive integer")
```
Comment (bmcfee) [inline librosa/sequence.py:1471]:
```suggestion
        raise ParameterError(f"n_states={n_states} must be a positive integer")
```

File: librosa/sequence.py [hunk 12]
```diff
@@ -1515,8 +1515,8 @@ def transition_loop(n_states, prob):
            [0.375, 0.375, 0.25 ]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 1:
-        raise ParameterError("n_states={} must be a positive integer > 1")
+    if not (is_positive_int(n_states) and (n_states > 1)):
+        raise ParameterError(f"n_states={} must be a positive integer > 1")
```
Comment (bmcfee) [inline librosa/sequence.py:1519]:
```suggestion
        raise ParameterError(f"n_states={n_states} must be a positive integer > 1")
```

File: librosa/sequence.py [hunk 13]
```diff
@@ -1582,8 +1582,8 @@ def transition_cycle(n_states, prob):
            [0.1, 0. , 0. , 0.9]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 1:
-        raise ParameterError("n_states={} must be a positive integer > 1")
+    if not (is_positive_int(n_states) and n_states > 1):
+        raise ParameterError(f"n_states={} must be a positive integer > 1")
```
Comment (bmcfee) [inline librosa/sequence.py:1586]:
```suggestion
        raise ParameterError(f"n_states={n_states} must be a positive integer > 1")
```

File: librosa/sequence.py [hunk 14]
```diff
@@ -1683,8 +1683,8 @@ def transition_local(n_states, width, *, window="triangle", wrap=False):
            [0.   , 0.   , 0.   , 0.   , 1.   ]])
     """
 
-    if not isinstance(n_states, (int, np.integer)) or n_states <= 1:
-        raise ParameterError("n_states={} must be a positive integer > 1")
+    if not (is_positive_int(n_states) and n_states > 1):
+        raise ParameterError(f"n_states={} must be a positive integer > 1")
```
Comment (bmcfee) [inline librosa/sequence.py:1687]:
```suggestion
        raise ParameterError(f"n_states={n_states} must be a positive integer > 1")
```

**Pull request merged**